src/logics/stargate-v2/logic.swap-token.ts

```diff
@@ -0,0 +1,129 @@
+import { StargatePool__factory } from './contracts';
+import * as common from '@protocolink/common';
+import * as core from '@protocolink/core';
+import { getDestChainIds, getEndpointId, getMarkets, getPoolByTokenAddress, supportedChainIds } from './configs';
+import { getNativeToken } from '@protocolink/common';
+import { utils } from 'ethers';
+
+export type SwapTokenLogicTokenList = {
+  srcToken: common.Token;
+  destChainIds: number[];
+}[];
+
+export type SwapTokenLogicParams = core.TokenToTokenExactInParams<{
+  receiver: string;
+}>;
+
+export type SwapTokenLogicFields = core.TokenToTokenExactInFields<{
+  receiver: string;
+  fee: string;
+  lzTokenFee: string;
+}>;
+
+export type SwapTokenLogicOptions = Pick<core.GlobalOptions, 'account'>;
+
+export class SwapTokenLogic extends core.Logic implements core.LogicBuilderInterface {
+  static id = 'swap-token';
+  static protocolId = 'stargate-v2';
+  static readonly supportedChainIds = supportedChainIds;
+
+  async getTokenList() {
+    const tokenList: SwapTokenLogicTokenList = [];
+    const srcTokens = [];
+
+    const markets = getMarkets(this.chainId);
+    for (const market of markets) {
+      srcTokens.push(market.token);
+    }
+
+    // find destination chain ids
+    for (const srcToken of srcTokens) {
+      const destChainIds = getDestChainIds(this.chainId, srcToken);
+      tokenList.push({ srcToken, destChainIds });
+    }
+
+    return tokenList;
+  }
```

Comment:
Wondering if we should provide `destTokens` to `getTokenList`, since we might need to display the destination token on the frontend. We need to pass the output token to the `quote` and `build` functions, but currently, `getTokenList` only returns the `chainId`.

Reply:
Stargate V2 bridges the same token across chains (say USDC from chainA to chainB). In this case, the output token should be the same as the input token. Will it be better to provide the output token info even if it is the same as the input token?